rest/src/main/java/com/redhat/ipaas/rest/v1/controller/handler/rtdb/RealTimeDBHandler.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.rest.v1.controller.handler.rtdb;
+
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+/**
+ * Provides a REST API to read/update a Key/Value database presented to the user a
+ * single large persistent JSON tree/document.  You can read or update subsets of the
+ * the document by specifying the path to the nodes in the JSON document that your
+ * interested in working with.
+ *
+ * This API aims to be very similar Firebase REST API:
+ * https://firebase.google.com/docs/database/rest/start
+ */
+@Api(value = "rtdb")
+@Path("/rtdb")
+@Component
+public class RealTimeDBHandler {
+    public static final String APPLICATION_JAVASCRIPT = "application/javascript";
+    private final RealTimeDB rtdb;
+
+    public RealTimeDBHandler(@Autowired RealTimeDB rtdb) {
+        this.rtdb = rtdb;
+    }
+
+    @Produces({APPLICATION_JSON, APPLICATION_JAVASCRIPT})
+    @Path("/{path: .*}.json")
+    @GET
+    public Response get(
+        @PathParam("path") String path,
+        @QueryParam("print") String print,
+        @QueryParam("shallow") Boolean shallow,
+        @QueryParam("callback") String callback
+    ) {
+        ImmutableGetOptions.Builder optionsBuilder = GetOptions.builder();
+        if ("pretty".equals(print)) {
+            optionsBuilder.prettyPrint(true);
+        } else if ("silent".equals(print)) {
+            if( rtdb.exists(path) ) {
+                return Response.noContent().build();
+            } else {
+                return Response.status(Response.Status.NOT_FOUND).build();
+            }
+        }
+        if( shallow!=null ) {
```

Comment:
shallow is not yet implemented.   see: https://firebase.google.com/docs/database/rest/retrieve-data#shallow